providers/fips/fipsprov.c

```diff
@@ -350,6 +350,7 @@ static const OSSL_ALGORITHM fips_kdfs[] = {
     { OSSL_KDF_NAME_SSKDF, "fips=yes", kdf_sskdf_functions },
     { OSSL_KDF_NAME_PBKDF2, "fips=yes", kdf_pbkdf2_functions },
     { OSSL_KDF_NAME_TLS1_PRF, "fips=yes", kdf_tls1_prf_functions },
+    { OSSL_KDF_NAME_KBKDF, "fips=yes", kdf_kbkdf_functions },
```

Comment:
I don't think we should be bringing this inside the FIPS module without a very strong reason. We've already agreed the scope of the FIPS module and the algorithms in it - so I'm not sure we can ask the lab to test more than that.

Reply:
It is possible for this to go into the fips module - it might just be vendor affirmed (meaning the lab would not test).

Reply:
@mattcaswell the reason for this work is to attempt to have krb5 be fips compliant, we contacted our lab and they determined this KDF is compliant, but we can apply downstream patches if necessary.

Also note that we will be submitting a few more KDFs (the older rfc3961 derive key, and one used by pkinit), so it is possible I will pull this work into a larger PR so it can be considered all at once.